src/components/ToDoPriority.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div
+    class="w-full h-full absolute bg-black opacity-10 rounded-xl left-0 top-0"
+    v-if="isDropDownShown"
+  ></div>
+  <div class="relative flex flex-wrap flex-col rounded-2xl">
+    <div class="flex justify-end">
+      <button
+        @click="showDropDown"
+        class="flex justify-center relative bg-white rounded-2xl sm:w-[125px] sm:h-[33px] w-[10px] h-[10px] sm:visible invisible"
+        :class="color(priority)"
+      >
+        {{ priority }} <img />
+        <div
+          v-if="isDropDownShown"
+          id="myDropdown"
+          class="absolute dropdown-content flex flex-col text-left w-[125px] top-[100%] pt-3"
+        >
+          <div
+            class="text-black border-black-500 border-2 bg-white rounded-2xl p-[17px]"
+          >
+            <p class="" @click="setPriority('High')">High</p>
+            <p class="" @click="setPriority('Medium')">Medium</p>
+            <p class="" @click="setPriority('Low')">Low</p>
+          </div>
+        </div>
+      </button>
+      <!-- {{ priority }}
+        <img class="w-[14px]" :src="VectorDown" /> -->
+    </div>
+  </div>
+</template>
+
+<script setup>
+  import { ref } from 'vue';
+  import VectorDown from './../assets/VectorDown.svg';
+
+  defineProps({ priority: { type: String } });
+
+  const emit = defineEmits(['update:priority']);
+
+  const isDropDownShown = ref(false);
+
+  function color(priority) {
```

Comment:
you can make this a computed variable instead of a function, which would make it easier to use.